src/core/main.c

```diff
@@ -1753,6 +1756,27 @@ static void update_cpu_affinity(bool skip_setup) {
                 log_warning_errno(errno, "Failed to set CPU affinity: %m");
 }
 
+static void update_numa_policy(bool skip_setup) {
+        int r;
+        _cleanup_free_ char *nodes = NULL;
+        const char * policy = NULL;
+
+        if (skip_setup || !mpol_is_valid(arg_numa_policy.type))
+                return;
+
+        if (DEBUG_LOGGING) {
+                policy = mpol_to_string(arg_numa_policy.type);
+                nodes = cpu_set_to_range_string(&arg_numa_policy.nodes);
+                log_debug("Setting NUMA policy to %s, with nodes %s.", strnull(policy), strnull(nodes));
+        }
+
+        r = apply_numa_policy(&arg_numa_policy);
+        if (r == -EOPNOTSUPP)
+                log_debug("NUMA support not available, ignoring.");
```

Comment:
if you have an error code, always pass it to the logging function, regardless if you include it in the msg text or not. hence:

`log_debug()` → `log_debug_errno(r,` in this case